WordPress/Classes/Services/MediaSettings.swift

```diff
@@ -3,14 +3,70 @@ import AVFoundation
 
 class MediaSettings: NSObject {
     // MARK: - Constants
+    fileprivate let imageOptimizationKey = "SavedImageOptimizationSetting"
     fileprivate let maxImageSizeKey = "SavedMaxImageSizeSetting"
+    fileprivate let imageQualityKey = "SavedImageQualitySetting"
     fileprivate let removeLocationKey = "SavedRemoveLocationSetting"
     fileprivate let maxVideoSizeKey = "SavedMaxVideoSizeSetting"
+    fileprivate let advertiseImageOptimizationKey = "SavedAdvertiseImageOptimization"
 
+    fileprivate let defaultImageOptimization = true
+    fileprivate let defaultMaxImageSize = 2000
+    fileprivate let defaultImageQuality: ImageQuality = .medium
+    fileprivate let defaultMaxVideoSize: VideoResolution = .sizeOriginal
+    fileprivate let defaultRemoveLocation = true
 
     fileprivate let minImageDimension = 150
     fileprivate let maxImageDimension = 3000
 
+    enum ImageQuality: String {
+        case maximum = "MaximumQuality100"
+        case high = "HighQuality90"
+        case medium = "MediumQuality80"
+        case low = "LowQuality70"
+
+        var doubleValue: Double {
+            switch self {
+            case .maximum:
+                return 1.0
+            case .high:
+                return 0.9
+            case .medium:
+                return 0.8
+            case .low:
+                return 0.7
+            }
+        }
+
+        var description: String {
+            switch self {
+            case .maximum:
+                return NSLocalizedString("Maximum", comment: "Indicates an image will use maximum quality when uploaded.")
+            case .high:
+                return NSLocalizedString("High", comment: "Indicates an image will use high quality when uploaded.")
+            case .medium:
+                return NSLocalizedString("Medium", comment: "Indicates an image will use medium quality when uploaded.")
+            case(.low):
+                return NSLocalizedString("Low", comment: "Indicates an image will use low quality when uploaded.")
+            }
+        }
+
+        static func imageQuality(from value: Float) -> MediaSettings.ImageQuality {
+            switch value {
+            case 1.0:
+                return .maximum
+            case 0.9:
+                return .high
+            case 0.8:
+                return .medium
+            case 0.7:
+                return .low
+            default:
+                return .medium
+            }
+        }
```

Comment:
You're right, it's not currently used. I created this function following the structure we have for the video resolution. However, since we won't use it for now we could remove it.

Reply:
Suggestion applied in https://github.com/wordpress-mobile/WordPress-iOS/pull/21981/commits/a85f4b11e14911ba81b0535d8e19f74ca25f8795.